fdbserver/RestoreMaster.actor.h

```diff
@@ -231,36 +236,39 @@ struct RestoreMasterData : RestoreRoleData, public ReferenceCounted<RestoreMaste
 			    .detail("NextRangeIndex", nextRangeIdx)
 			    .detail("UsedLogFiles", curLogFiles.size());
 
+			ASSERT(prevEndVersion < nextVersion); // Ensure progress
 			if (vb.size + nextVersionSize <= opConfig.batchSizeThreshold) {
 				// nextVersion should be included in this batch
-				bool advanced = false;
 				vb.size += nextVersionSize;
 				while (rangeIdx < nextRangeIdx) {
+					ASSERT(rangeFiles[rangeIdx].fileSize > 0);
 					vb.rangeFiles.push_back(rangeFiles[rangeIdx]);
 					++rangeIdx;
-					advanced = true;
 				}
 
 				for (auto& log : curLogFiles) {
 					ASSERT(log.beginVersion < nextVersion);
 					ASSERT(log.endVersion > prevEndVersion);
+					ASSERT(log.fileSize > 0);
 					vb.logFiles.push_back(log);
-					advanced = true;
 				}
 
-				ASSERT(advanced == true || nextVersionSize > 0); // Ensure progress
 				vb.endVersion = nextVersion;
 				prevEndVersion = vb.endVersion;
 			} else {
 				if (vb.size < 1) {
 					// [vb.endVersion, nextVersion) > opConfig.batchSizeThreshold. We should split the version range
 					if (prevEndVersion >= nextVersion) {
-						// If range files at one version > batchSizeThreshold, DBA should increase batchSizeThreshold
+						// If range files at one version > batchSizeThreshold, DBA should increase batchSizeThreshold to
+						// some value larger than nextVersion
 						TraceEvent(SevError, "FastRestoreBuildVersionBatch")
 						    .detail("NextVersion", nextVersion)
 						    .detail("PreviousEndVersion", prevEndVersion)
 						    .detail("NextVersionIntervalSize", nextVersionSize)
-						    .detail("BatchSizeThreshold", opConfig.batchSizeThreshold);
+						    .detail("BatchSizeThreshold", opConfig.batchSizeThreshold)
+						    .detail("SuggestedMinimumBatchSizeThreshold", nextVersion);
+						// Exit restore early if it won't succeed
+						flushAndExit(FDB_EXIT_ERROR);
```

Comment:
What process is running this code?

Reply:
fdbserver process runs as restore role.

Reply:
So this is going to kill a restore server process? I'm not sure how that will filter back to the person executing the restore, but my initial take is that this might be a little confusing. It will also probably be the case in many setups that the process will get quickly restarted; what effect would that have? Alternatively, would it be bad if the process wasn't restarted? Does this have any effect on the running restore operation (i.e. does it cancel it, etc)?

I think a cleaner approach would be to have a way to push this kind of information back through the restore status mechanism, which could potentially be left here as a TODO item in this PR if the overall feature is still a work in progress.

Reply:
This is a great point! The current implementation does not consider the death of one process. To restart a restore, all restore process should be bounced and the destination cluster should be cleared. 

I will mark this as a TODO.